pallet/staking/src/lib.rs

```diff
@@ -464,10 +458,6 @@ pub mod pallet {
 		pub fn collect(origin: OriginFor<T>, commission: Perbill) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			if commission > T::MaxCommission::get() {
-				Err(<Error<T>>::CommissionTooHigh)?;
-			}
-
 			<Collators<T>>::mutate(&who, |c| *c = Some(commission));
```

Comment:
Should we check the commission not bigger than 100% and less than 0%?

Reply:
No need to do that, the type `Perbill` type adds such restrictions to the input value. 

```sh
2023-10-12 13:45:49 panicked at 'Bad input data provided to query_info: Codec error', /home/bear/coding/rust-space/darwinia/runtime/pangolin/src/lib.rs:234:1    
2023-10-12 13:45:50 panicked at 'Bad input data provided to validate_transaction: Codec error', /home/bear/coding/rust-space/darwinia/runtime/pangolin/src/lib.rs:234:1 
```
You transaction with invalid commissions as you said will be rejected by the runtime.